### ISSUE TITLE
Limit filesystem access under Android 6.0

### ISSUE THREAD

**Issue by wb7** [M66B/XPrivacy#2371]
Hi everyone,

I've installed CM13/Android 6.0.1 today on my klte (Samsung Galaxy S5), but XPrivacy didn't work as expected and known from Lollipop. If I limit Whatsapps "open" permission and set it to "ask", but nothing else on the storage-category, then usually XPrivacy asked for permission for the specific path Whatsapp tried to access - for example, if I selected "make backup" in Whatsapp, XPrivacy asked if Whatsapp should access the backup folder. This is not the case since I am using Android 6.0. The only permission from "storage", which is used while making the backup (according to the last-use-timestamp) is getExternalStorageState.

Why can't I limit the filesystem paths under Android 6.0? Is there any workaround or am I doing something wrong?

Thanks in advance!

**Comment by M66B** [comment]
Please provide a logcat with XPrivacy debugging enabled

**Comment by CloudyReimu** [comment]
This problem do exist in Android 6.0, as I confirmed it just now.

Device: Asus Zenfone 2

OS: CyanogenMod 13 latest nightly (20161219)

Xposed: Latest (v87, API 23)

Procedure:
1.Turn on all debug/logging settings in XPrivacy
2.In XPrivacy, set the storage permission of an app (I used Plus Messenger as an example) to 'ask' (show as question marks in checkboxes).
3.Open the Plus Messenger app and try to send a file.
4. XPrivacy will ask for the 'getExternalStorageState' permission but will NEVER ask the 'open' permission.
5.As a result, browsing files in the in-app file explorer is fully unrestricted.

Screenshots:
https://goo.gl/photos/bCkoNstktikfXgZU8

Logcat file:
https://drive.google.com/file/d/0B2eNTCyy3IPQVFRoM1hHYzFvUlE/view?usp=drivesdk

Thanks!
